src/components/SessionExpiry.jsx

```diff
@@ -34,7 +33,9 @@ const SessionExpiry = () => {
       setIsLoggedIn(false)
       setTimeLeft(0)
     }
-  }, [isLoggedIn, timeLeft, isReset]);
+  }
+
+  useEffect(autoLogOut);
```

Comment:
so when autoLogOut callback is initiated is it executed in react like `return autoLogOut()` which returns another callback viz `countDown` ?

Reply:
So as far as autoLogOut.  I think it might be best to use useCallback hook and pass in isLoggedIn as a dependency.  So that autoLogOut only gets called when isLoggedIn changes to false.  But I think that the isLoggedIn and the autoLogOut by best be held in the state of where the Authentication takes place and Seesion Expiry Modal should be a child of that component and receive them as props.  Does that make sense?